store/tikv/region_cache.go

```diff
@@ -1765,13 +1793,16 @@ const (
 	deleted
 )
 
-// initResolve resolves addr for store that never resolved.
+// initResolve resolves the address of the store that never resolved and returns an
+// emptry string if it's a tombstone.
 func (s *Store) initResolve(bo *Backoffer, c *RegionCache) (addr string, err error) {
 	s.resolveMutex.Lock()
 	state := s.getResolveState()
 	defer s.resolveMutex.Unlock()
 	if state != unresolved {
-		addr = s.addr
+		if state != deleted {
+			addr = s.addr
+		}
```

Comment:
In L1843, should we also check if `state` is `deleted` and return empty addr?

Reply:
It's impossible there are concurrent `initResolve` because of `s.resolveMutex.Lock()`, but I find there is a race between ` asyncCheckAndResolveLoop` and `initResolve`. I will fix it.